sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -575,19 +575,20 @@ object GpuOverrides extends Logging {
     lit.value == null
   }
 
-  def isNullOrEmptyOrRegex(exp: Expression): Boolean = {
+  def isSupportedStringReplacePattern(exp: Expression): Boolean = {
     val lit = extractLit(exp)
     if (!isOfType(lit, StringType)) {
-      false
+      true
```

Comment:
Curious, when do we expect the replace pattern to not be a string?  Do we want to say it is supported if it is not a string?

Reply:
This is a confusing check and as far as I know, this will always be a literal UTF8String. I have now simplified this method to use pattern matching.